fsspec/implementations/local.py

```diff
@@ -232,10 +248,20 @@ def chmod(self, path, mode):
         return os.chmod(path, mode)
 
 
-def make_path_posix(path, sep=os.sep):
+def make_path_posix(path, sep=os.sep, remove_trailing_slash=False):
     """Make path generic"""
     if isinstance(path, (list, set, tuple)):
-        return type(path)(make_path_posix(p) for p in path)
+        return type(path)(
+            make_path_posix(p, remove_trailing_slash=remove_trailing_slash)
+            for p in path
+        )
+    path = _make_path_posix(path, sep)
+    if os.name == "nt" and len(path) == 3 and path[1] == ":":
+        return path  # windows root
+    return path.rstrip("/") if remove_trailing_slash else path
```

Comment:
Presumably this i what breaks the mapper test; the key in question I think would have hit "if path.startswith("/"):" previously.

Reply:
>> The trouble here is, that for real filesystems (local, ftp, ssh, ...), files cannot e d with "/", even though directory names are sometimes printed with the extra character. 

The filename ends with a trailing '/'. Is it supposed to?



![image](https://github.com/fsspec/filesystem_spec/assets/1474668/9d07edc2-0be6-4815-94a8-311d1b8ecfde)

I should mention that I also added that check in open, however the issue still occurs if the check is disabled.

The issue can be resolved by stripping the trailing suffix in` mapping.py->__getitem__`.   Doing doesn't appear to cause any regression.